src/Distributed/distributed_grids.jl

```diff
@@ -121,13 +151,18 @@ function reconstruct_global_grid(grid::RectilinearGrid)
 
     TX, TY, TZ = topology(grid)
 
+    TX = reconstruct_global_topology(TX, Rx, ri, arch.communicator)
+    TY = reconstruct_global_topology(TY, Ry, rj, arch.communicator)
+    TZ = reconstruct_global_topology(TZ, Rz, rk, arch.communicator)
+
     x = cpu_face_constructor_x(grid)
     y = cpu_face_constructor_y(grid)
     z = cpu_face_constructor_z(grid)
 
-    xG = assemble(x, nx, Rx, i, arch)
-    yG = assemble(y, ny, Ry, j, arch)
-    zG = assemble(z, nz, Rz, k, arch)
+    ## This will not work with 2D parallelizations!!
+    xG = Rx == 1 ? x : assemble(x, nx, Rx, ri, arch)
+    yG = Ry == 1 ? y : assemble(y, ny, Ry, rj, arch)
+    zG = Rz == 1 ? z : assemble(z, nz, Rz, rk, arch)
```

Comment:
the names `xG` etc are potentially confusing to people who know MITgcm because they have a meaning there (totally unrelated to this one).